miner/worker.go

```diff
@@ -1048,15 +1016,24 @@ func (w *worker) fillTransactions(interruptCh chan int32, env *environment) {
 			localTxs[account] = txs
 		}
 	}
+
+	var stopTimer *time.Timer
+	delay := w.engine.Delay(w.chain, env.header, &w.config.DelayLeftOver)
+	if delay != nil {
+		stopTimer = time.NewTimer(*delay)
+		log.Debug("Time left for mining work", "delay", delay.String())
+		defer stopTimer.Stop()
+	}
+
 	if len(localTxs) > 0 {
 		txs := types.NewTransactionsByPriceAndNonce(env.signer, localTxs, env.header.BaseFee)
-		if w.commitTransactions(env, txs, interruptCh) {
+		if w.commitTransactions(env, txs, interruptCh, stopTimer) {
```

Comment:
```
	x := time.NewTimer(time.Second)
	<-x.C
	fmt.Println("read delay 1")
	<-x.C
	fmt.Println("read delay 2")
```

It turns out that `fmt.Println("read delay 2")` is not reachable. 

So if the timer trigger when `commitTransactions` for local tx, then dead lock will happen.